tests/unit/s2n_pq_kem_test.c

```diff
@@ -47,6 +48,8 @@ int main()
 
     for (size_t i = 0; i < s2n_array_len(test_vectors); i++) {
         const struct s2n_kem *kem = test_vectors[i];
+        bool expect_success = s2n_libcrypto_supports_evp_kem()
+                && ((kem != &s2n_mlkem_768) || s2n_libcrypto_supports_mlkem());
```

Comment:
This looks right to me, but it's harder to reason about than I'd like. Instead of just not asserting when you don't expect success, it would be safer to branch below and assert that when you don't expect mlkem, the test fails as expected. That way you don't accidentally silently skip more than you intended to.

Reply:
Fixed. Consolidated logic into a new `s2n_kem_is_available()` API.